### PR TITLE
chore(common): fix solana and ada support

### PR DESCRIPTION
**Pull request by tomasklim**
During update of T3T1 support in our suite repo, I found that Solana was hardcoded into our `coins.json` list and not generated from common

- added solana support
- fixed cardano suite support
- removed unused `coins_details.json` file

### CODE REVIEW

**Review by matejcik — APPROVED** [review]

**Comment by matejcik** [comment]
> `common/defs/coins_details.json` is updated by hand. I haven't found a way how to update it automatically

while you're touching this, please make a separate commit that removes this file. it should not be in firmware anymore

**Comment by tomasklim** [comment]
done

**Comment by matejcik** [comment]
perfect, thanks. waiting for CI before merging.

**Pull request merged**
